dev/com.ibm.ws.session.db/src/com/ibm/ws/session/store/db/DatabaseStoreService.java

```diff
@@ -168,21 +169,19 @@ protected void deactivate(ComponentContext context) {
         dataSourceFactoryRef.deactivate(context);
         uowCurrentRef.deactivate(context);
         userTransactionRef.deactivate(context);
-        if (isCompletedPassivation()) { //START 128284
-            serializationServiceRef.deactivate(context);
-        } else {
-
-            while (!isCompletedPassivation()){
-                try {
-                    Thread.sleep(100L); // sleep 1/10th of a second
-                } catch (InterruptedException e) {
-                    FFDCFilter.processException(e, this.getClass().getName(), "180");
-                } finally {
-                    serializationServiceRef.deactivate(context);
-                }
+        
+        // Block the progress of deactivate so that session manager is able to access the DB until it finishes stopping applications.
+        // The approach of blocking is aligned CacheStoreService with MAX_WAIT
+        final long MAX_WAIT = TimeUnit.SECONDS.toNanos(20);
+        for (long start = System.nanoTime(); !completedPassivation && System.nanoTime() - start < MAX_WAIT;)
```

Comment:
Nit-pick: Add curly brackets around the try/catch statement so that the loop body is easier to read

Reply:
Thanks